File: app/src/main/java/com/incendiary/android/data/DataManager.java
package com.incendiary.android.data;

import java.util.List;
import javax.inject.Inject;
import javax.inject.Singleton;
import rx.Observable;
import com.incendiary.android.data.local.DatabaseHelper;
import com.incendiary.android.data.model.Ribot;
import com.incendiary.android.data.remote.ApiService;

@Singleton public class DataManager {

  private final ApiService mApiService;
  private final DatabaseHelper mDatabaseHelper;

  @Inject public DataManager(ApiService apiService, DatabaseHelper databaseHelper) {
    mApiService = apiService;
    mDatabaseHelper = databaseHelper;
  }

  public Observable<Ribot> syncRibots() {
    return mApiService.getRibots().concatMap(mDatabaseHelper::setRibots);
  }

  public Observable<List<Ribot>> getRibots() {
    return mDatabaseHelper.getRibots().distinct();
  }
}

File: app/src/main/java/com/incendiary/androidboilerplate/data/DataManager.kt
package com.incendiary.androidboilerplate.data

import com.incendiary.androidboilerplate.data.local.DatabaseHelper
import com.incendiary.androidboilerplate.data.model.Ribot
import com.incendiary.androidboilerplate.data.remote.ApiService
import rx.Observable
import javax.inject.Inject
import javax.inject.Singleton

@Singleton class DataManager
@Inject constructor(private val mApiService: ApiService, private val databaseHelper: DatabaseHelper) {

	fun syncRibots(): Observable<Ribot> {
		return mApiService.getRibots().concatMap { databaseHelper.setRibots(it) }
	}

	val ribots: Observable<List<Ribot>>
		get() = databaseHelper.ribots.distinct()
}
